### PR TITLE
feat: create line profile page

### PR DESCRIPTION
**Pull request by VSciFlight**
# Description
Added a profile page accessible only by the path of '/profile'

- it has the inputs to determine the line the user intends to see
- an empty timetable
- presumably a timeline graph of that bus comparing the schedule against reality (now it is recommended to see it as template).

### things left to do:
- integrate the inputs to the profile so that the profile shows the info according to the line it received
- fixed timetable that fetch the timetable for that line
- graph shows relevant information

## screenshots
image

### CODE REVIEW

File: src/pages/Profile.tsx [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+
+const LineProfileComponent = () => {
+  const { search, setSearch } = useContext(SearchContext)
+  const { operatorId, lineNumber, timestamp, routes, routeKey } = search
+  const [agencyList, setAgencyList] = useState<Agency[]>([])
+
+  return (
+    <Grid xs={12} lg={6}>
+      <div className="widget">
+        <h2 className="title">{TEXTS.profile_page}</h2>
```
Comment (NoamGaash) [inline src/pages/Profile.tsx:107]:
```suggestion
        <h2 className="title">{t('profile_page')}</h2>
```

Comment (NoamGaash) [reply src/pages/Profile.tsx:107]: text should be from: https://github.com/hasadna/open-bus-map-search/blob/33e83cf2747b10c6d86dfbcb23fb368427dbbabd/src/locale/en.json#L4

File: src/pages/components/BusArrivalsTimeline.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+{/* Series 1 indicates the schedule */}
+{/* Series 2 indicates the reality */}
+{/*  */}
+
+
+{/* NEEDS:
+ - timestamp and station data from the schedule
+ - timestamp and station data from the reality
+ - graph
+*/}
```
Comment (NoamGaash) [inline src/pages/components/BusArrivalsTimeline.tsx:24]: open as issue

File: src/pages/components/BusArrivalsTimeline.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    Legend,
+    ResponsiveContainer,
+  } from 'recharts'
+
+
+{/* Timeline Graph shows a graph which its X-axis is time or station and Y-axis is time passed (in minutes) */}
+{/* The graph follows the progress of a bus and tracks its correspondence to the schedule */}
+{/* Series 1 indicates the schedule */}
+{/* Series 2 indicates the reality */}
+{/*  */}
```
Comment (NoamGaash) [inline src/pages/components/BusArrivalsTimeline.tsx:17]: move to a paragraph (make it visible)

File: src/pages/components/BusArrivalsTimeline.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+            {arrayGroup(data, (item) => item.id)
+            .filter((group) => group.length > 1 && group.reduce((a, b) => a + b.current, 0) > 1)
+            .map((group) => (
+                <div key={group[0].name}>
+                <h3 className="title">{group[0].name}</h3>
+                <ResponsiveContainer width="100%" height={300}>
+                    <LineChart
+                    data={group
+                        .sort((a, b) => (a.gtfs_route_date > b.gtfs_route_date ? 1 : -1))
+                        .sort((a, b) => (a.gtfs_route_hour > b.gtfs_route_hour ? 1 : -1))}
```
Comment (NoamGaash) [inline src/pages/components/BusArrivalsTimeline.tsx:63]: I believe it's wrong
consider 1.1.2000 15:00 vs 2.2.2000 14:00

File: src/pages/components/BusArrivalsTimeline.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+      current: number
+      max: number
+      percent: number
+      gtfs_route_date: string
+      gtfs_route_hour: string
+    }[]
+  })
+  {
+    return (
+        <div className="chart">
```
Comment (NoamGaash) [inline src/pages/components/BusArrivalsTimeline.tsx:53]: please split to smaller sub components

**Review by NoamGaash — CHANGES_REQUESTED** [review]
Good work :pray:

File: src/pages/components/BusArrivalsTimeline.tsx (hunk 2 above)
Comment (VSciFlight) [reply src/pages/components/BusArrivalsTimeline.tsx:24]: opened :)
https://github.com/hasadna/open-bus-map-search/issues/220

File: src/pages/components/BusArrivalsTimeline.tsx (hunk 3 above)
Comment (VSciFlight) [reply src/pages/components/BusArrivalsTimeline.tsx:17]: done inside the graph container box

File: src/pages/components/BusArrivalsTimeline.tsx (hunk 4 above)
Comment (VSciFlight) [reply src/pages/components/BusArrivalsTimeline.tsx:63]: this data is irrelevant to the page and was taken from the dashboard page.
looking to update it in the future, I wanted a template to base the new graph

File: src/resources/texts.tsx [hunk 6]
```diff
@@ -183,6 +185,7 @@ export const TEXTS = {
   kmh: 'קמ״ש',
   profile_page: 'דף פרופיל',
   profile_page_line: 'דף פרופיל על קו',
+  github_link: 'github_link',
```
Comment (NoamGaash) [inline src/resources/texts.tsx:188]: why?

File: src/api/groupByService.ts [hunk 7]
```diff
@@ -63,6 +63,14 @@ export type GroupByRes = Replace<
   | undefined
 >
 
+export type GroupByX = {
```
Comment (NoamGaash) [inline src/api/groupByService.ts:66]: is it used?

File: src/pages/Profile.tsx [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+  // const resp = getGtfsRidesList(new Date(), operator, lineNumber, routeId)
+  const resp = getGtfsRidesList(new Date(), '3', '271', '1')
+
+  return (
+    <Grid xs={12} lg={6}>
+      <div className="widget">
+        <h2 className="title">{t('profile_page')}</h2>
+        {/* <label> מפעיל: {operator} </label>
+        <label> מספר קו: {line} </label>
+        <label> כיוון נסיעה: {route} </label> */}
```
Comment (NoamGaash) [inline src/pages/Profile.tsx:99]: remove please

**Review by NoamGaash — APPROVED** [review]

**Comment by shootermv** [comment]
i need some explanation please, if i want to access the profile page, where should be a link leading to it?

**Comment by VSciFlight** [comment]
@shootermv there isn't a link at this point due work in progress. you can access the profile page at "/profile"
image

**Comment by NoamGaash** [comment]
@VSciFlight have you seen the linter output?
thanks

**Comment by NoamGaash** [comment]
image
you can also see it inside github

**Comment by VSciFlight** [comment]
So, I have made changes to the page and left the modules that don't crash (for now) or having issues with them.

**Pull request merged**